### ISSUE TITLE
False positive for HOSTNAME if used as a variable

### ISSUE THREAD

**Issue by richq** [koalaman/shellcheck#354]
The following snippet gives the warning `test.sh:4:7: warning: In POSIX sh, HOSTNAME is not supported. [SC2039]`

```
#!/bin/sh

HOSTNAME=$(hostname)
echo "$HOSTNAME"
```

Seems like this comes from the bashism checks added in cc86aab3f12130589d0d9a4f4cc514513050b5dd. Does shellcheck have the information available to know that HOSTNAME was (re)defined by the script prior to being used? It'd be good to not show that as a warning in this case.

**Comment by bittorf** [comment]
me too, this does not make sense. even this fails:

```
#!/bin/sh
export HOSTNAME=bla
echo "$HOSTNAME"
```

**Comment by koalaman** [comment]
Fixed in 2488be7

**Comment by richq** [comment]
Very nice, thank you!
